shell/common/engine_unittests.cc

```diff
@@ -299,4 +302,44 @@ TEST_F(EngineTest, PassesLoadDartDeferredLibraryErrorToRuntime) {
   });
 }
 
+TEST_F(EngineTest, NotifyIdleMoreThanOncePerFiveSeconds) {
```

Comment:
Don't you want to show a timedelta that is less that 5 seconds so you can show it doesn't get called?

Reply:
I'm not sure I have it clear in my head about which case you're asking for.

This case is covering three calls, at 10s, 13s, and 16s. We're verifying that NotifyIdle only gets called twice: the very first call and the last one.